src/main/osd/osd_elements.c

```diff
@@ -578,13 +579,29 @@ static void osdElementEscTemperature(osdElementParms_t *element)
     }
 }
 
+#endif // USE_ESC_SENSOR
+
+#if defined(USE_ESC_SENSOR) || defined(USE_RPM_FILTER)
+int getEscRpmFreq(int i ) 
+{
+    if (motorConfig()->dev.useDshotTelemetry) {
+        return 100.0f / 60.0f  / (motorConfig()->motorPoleCount / 2.0f) * getDshotTelemetry(i);
+    } else {
+       return calcEscRpm(getEscSensorData(i)->rpm) / 60;
+ 
+    }  
+}
+
 static void osdElementEscRpm(osdElementParms_t *element)
 {
-        if (featureIsEnabled(FEATURE_ESC_SENSOR)) {
-            tfp_sprintf(element->buff, "%5d", osdEscDataCombined == NULL ? 0 : calcEscRpm(osdEscDataCombined->rpm));
-        }
+    for (int i=0; i < getMotorCount(); i++ ) {
+	char freqStr[4];
+	itoa(getEscRpmFreq(i),freqStr,10);
```

Comment:
This is now returning an incorrect value for RPM.

Reply:
You meant you want to display rpm not the rpm frequency?  I converted the reading from esc sensor and bidirectional dshot to frequency.

Reply:
Yes, the existing OSD element is displaying RPM (which is also the much more common unit), and there is no reason to change that.

Reply:
Ok will do the change. Thanks!

Reply:
This is now displaying rpm.  I did a test flight this morning. All good.
Here is the flight footage,  the telemetry reading is much more visible in the goggles than in the dvr feed. https://youtu.be/U3MxeY8mfmg